proposals/src/proposal/cosmwasm/anchor_update.rs

```diff
@@ -48,14 +42,20 @@ impl AnchorUpdateProposal {
 
     /// Get the source chain.
     #[must_use]
-    pub const fn src_chain(&self) -> TypedChainId {
-        self.src_chain
+    pub fn src_chain(&self) -> TypedChainId {
+        self.src_resource_id.typed_chain_id()
+    }
+
+    /// Get the src_resource_id identifier.
+    #[must_use]
+    pub const fn src_resource_id(&self) -> ResourceId {
+        self.src_resource_id
     }
 
     /// Get the latest leaf index.
     #[must_use]
     pub const fn latest_leaf_index(&self) -> u32 {
-        self.latest_leaf_index
+        self.header.nonce.to_u32()
```

Comment:
@drewstone 
one question, is there any guarantee that the `nonce` & `latest_leaf_index` matches all the time?
What if other proposals(like `treasury`) are interleaved?

Reply:
We are delegating the nonce check to the lowest level possible, so for example the flow of a Signed Proposal is like so

Signature Bridge -> Handler -> The Anchor/Tokens ..etc 

So the last level is when we check for the nonce, and in the Anchor update proposals we will use the nonce as the leaf index, so to answer your question, Yes they are both the same and you can just say there is no more leaf index it is just a nonce after all.

Reply:
Thanks for the detailed answer @shekohex